src/Command/ErrorFormatter/SummaryErrorFormatter.php

```diff
@@ -0,0 +1,56 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Command\ErrorFormatter;
+
+use PHPStan\Command\AnalysisResult;
+
+class SummaryErrorFormatter implements ErrorFormatter
+{
+
+	const UNSPECIFIED = 'unspecified';
+
+	public function formatErrors(
+		AnalysisResult $analysisResult,
+		\Symfony\Component\Console\Style\OutputStyle $style
+	): int
+	{
+		if (!$analysisResult->hasErrors()) {
+			return 0;
+		}
+
+		$errors = [];
+		$notFileSpecificErrorsCount = count($analysisResult->getNotFileSpecificErrors());
+		if ($notFileSpecificErrorsCount) {
+			$errors[self::UNSPECIFIED] = $notFileSpecificErrorsCount;
+		}
+
+		foreach ($analysisResult->getFileSpecificErrors() as $fileSpecificError) {
+			$key = $fileSpecificError->getRule() ?? self::UNSPECIFIED;
+			if (!array_key_exists($key, $errors)) {
+				$errors[$key] = 0;
+			}
+			$errors[$key]++;
+		}
+
+		uasort($errors, function ($a, $b) {
+			return $b <=> $a;
+		});
+
+		$padding = null;
+		foreach ($errors as $description => $count) {
+			if ($padding === null) {
+				$padding = (int) floor(log($count, 10)) + 1;
+			}
+			$style->write(
+				sprintf(
+					sprintf('%%\' %dd %%s', $padding), // "%' 1d %s"
```

Comment:
This is super-unreadable. Php unfortunately does not supports the `%*` syntax which works in C. How about using `str_pad` instead, i.e. sth like

~~~php
sprintf("%s %s\n", str_pad($count, $padding), $description);
~~~